From fb0aee67e031efb154c73b1140298b193a7fa7ef Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Fri, 13 Oct 2017 14:47:51 +0800
Subject: [PATCH 2635/5242] MLK-16584 ARM64: dts: imx8qm-mek: enable Murata
 1CQ wifi bt

commit  2558a954a3865e6c08f6fd378b988b217ff8c2dc from
https://source.codeaurora.org/external/imx/linux-imx.git

Add 1CQ wifi bt support for i.MX8QM MEK board.
- Support MEK onboard pcie0 M.2 interface, test pass on 1CQ wifi.
- Support HCI Uart interface, test pass on 1CQ bt.
- Add lpuart interface support for bt, console, mkbus.

BuildInfo:
 - SCFW d0458f9f, IMX-MKIMAGE 1c6fc7d8, ATF a438801
 - U-Boot 2017.03-00042-g543559e

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts |   83 +++++++++++++++++++++-
 1 file changed, 81 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index ccee6cb..bfd3f7b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -25,11 +25,28 @@
 		stdout-path = &lpuart0;
 	};
 
+	modem_reset: modem-reset {
+		compatible = "gpio-reset";
+		reset-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <2000>;
+		reset-post-delay-ms = <40>;
+		#reset-cells = <0>;
+	};
+
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		epdev_on: fixedregulator@100 {
+			compatible = "regulator-fixed";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-name = "epdev_on";
+			gpio = <&gpio4 9 0>;
+			enable-active-high;
+		};
+
 		reg_usdhc2_vmmc: usdhc2_vmmc {
 			compatible = "regulator-fixed";
 			regulator-name = "sw-3p3-sd1";
@@ -110,11 +127,34 @@
 			fsl,pins = <
 				SC_P_UART0_RX_DMA_UART0_RX		0x06000020
 				SC_P_UART0_TX_DMA_UART0_TX		0x06000020
-				SC_P_UART0_RTS_B_DMA_UART0_RTS_B	0x06000020
-				SC_P_UART0_CTS_B_DMA_UART0_CTS_B	0x06000020
 			>;
 		};
 
+		pinctrl_lpuart1: lpuart1grp {
+			fsl,pins = <
+				SC_P_UART1_RX_DMA_UART1_RX		0x06000020
+				SC_P_UART1_TX_DMA_UART1_TX		0x06000020
+				SC_P_UART1_CTS_B_DMA_UART1_CTS_B	0x06000020
+				SC_P_UART1_RTS_B_DMA_UART1_RTS_B	0x06000020
+				SC_P_QSPI1A_DQS_LSIO_GPIO4_IO22		0x00000021
+			>;
+		};
+
+		pinctrl_lpuart2: lpuart2grp {
+			fsl,pins = <
+				SC_P_UART0_RTS_B_DMA_UART2_RX		0x06000020
+				SC_P_UART0_CTS_B_DMA_UART2_TX		0x06000020
+			>;
+		};
+
+		pinctrl_lpuart3: lpuart3grp {
+			fsl,pins = <
+				SC_P_M41_GPIO0_00_DMA_UART3_RX		0x06000020
+				SC_P_M41_GPIO0_01_DMA_UART3_TX		0x06000020
+			>;
+		};
+
+
 		pinctrl_sai1: sai1grp {
 			fsl,pins = <
 				SC_P_SAI1_RXD_AUD_SAI1_RXD		0x0600004c
@@ -133,6 +173,16 @@
 			>;
 		};
 
+		pinctrl_pciea: pcieagrp{
+			fsl,pins = <
+				SC_P_PCIE_CTRL0_CLKREQ_B_LSIO_GPIO4_IO27	0x00000021
+				SC_P_PCIE_CTRL0_WAKE_B_LSIO_GPIO4_IO28		0x00000021
+				SC_P_PCIE_CTRL0_PERST_B_LSIO_GPIO4_IO29		0x00000021
+				SC_P_LVDS1_I2C0_SDA_LVDS1_GPIO0_IO03		0x00000021
+				SC_P_USDHC2_RESET_B_LSIO_GPIO4_IO09		0x00000021
+			>;
+		};
+
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
 				SC_P_EMMC0_CLK_CONN_EMMC0_CLK		0x06000041
@@ -335,6 +385,25 @@
 	status = "okay";
 };
 
+&lpuart1 { /* BT */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart1>;
+	resets = <&modem_reset>;
+	status = "okay";
+};
+
+&lpuart2 { /* Dbg console */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart2>;
+	status = "okay";
+};
+
+&lpuart3 { /* MKbus */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart3>;
+	status = "okay";
+};
+
 &gpu_3d0 {
         status = "okay";
 };
@@ -346,3 +415,13 @@
 &imx8_gpu_ss {
         status = "okay";
 };
+
+&pciea{
+	ext_osc = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pciea>;
+	disable-gpio = <&gpio0 3 GPIO_ACTIVE_LOW>;
+	reset-gpio = <&gpio4 29 GPIO_ACTIVE_LOW>;
+	epdev_on-supply = <&epdev_on>;
+	status = "okay";
+};
-- 
1.7.9.5

